src/coreclr/tools/Common/JitInterface/ThunkGenerator/InstructionSetGenerator.cs

```diff
@@ -251,14 +251,6 @@ public bool ParseInput(TextReader tr)
                 }
             }
 
-            foreach (var architectureInfo in _architectureJitNames)
-            {
-                if (architectureInfo.Value.Count > 62)
-                {
-                    throw new Exception("Too many instruction sets added. Scheme of using uint64_t as instruction mask will need updating");
-                }
-            }
```

Comment:
I think we should keep this in. We could declare a:

```cpp
            // This represents the number of flags fields we currently track
            const int FlagsFieldCount = 1;
```

and then have this check be:
```cpp
            foreach (var architectureInfo in _architectureJitNames)
            {
                if (architectureInfo.Value.Count > ((FlagsFieldCount * sizeof(ulong) * 8) - 2))
                {
                    throw new Exception("Too many instruction sets added. Update FlagsFieldCount");
                }
            }
```

That makes it really straightforward to "update" and ensures that the current configuration stays inline.

Depending on what @davidwrighton says, we may also need to preserve bit 63 as `NONE`, in which case this check would also help handle that scenario

Reply:
Added a private field FlagsFieldCount and using it